test/invoker_test.rb

```diff
@@ -171,7 +171,7 @@
         warnings = err.string
       end
       assert_equal false, $VERBOSE
-      refute_empty warnings
+      refute_empty warnings unless RUBY_ENGINE == 'truffleruby'
```

Comment:
Could we assert what the warnings are when running on TruffleRuby?

Reply:
`warnings` is empty. I guess that `-w` has no effect when running on TruffleRuby.
You want to add the following assertion:

```rb
if RUBY_ENGINE == 'truffleruby'
  assert_empty warnings # no script warnings when running on TruffleRuby
else
  refute_empty warnings
end
```

Reply:
Yes please. Let's assert the current behavior so we know when it changes.

Reply:
`-w` enables warnings on TruffleRuby like on MRI, but maybe the particular warning you're expecting here is not emitted. Although in this case I guess `-w` is passed to AsciiDoctor, not as a Ruby interpreter option?

Reply:
Asciidoctor passes the option through.

Reply:
We turn on script warnings by setting `$VERBOSE = true`. Perhaps this doesn't work on TruffleRuby.

Reply:
Aha! Here's the problem. The following code doesn't generate a warning in TruffleRuby.

```rb
puts $NO_SUCH_VARIABLE
```

However, the following does:

```rb
def fn *args
end

vals = %w(a b c)
fn *vals
```

So we just need a different example.

Reply:
I'd like the find the most concise code that can produce a warning. There are lots of other things that could work.

Reply:
Wow, TruffleRuby really doesn't complain about much. Most of the warnings I know of in Ruby don't show up in TruffleRuby.

Reply:
Got it:

```rb
String /foo/
```

Reply:
What does this code do ? 🤔

Reply:
We are validating that using `-w` turns on Ruby script warnings. So we need to make a warning happen.

Reply:
Yes but I'm curious since this code is "valid" Ruby, what does it do ?

Reply:
It converts the regular expression to a String. But it throws a warning since the `/` is ambiguous.

Reply:
I see... thanks for the explanation. I will go ahead and replace the code to make sure that a warning is thrown when running on TruffleRuby, MRI or JRuby.

Reply:
Thanks!

If you want, you can leave a comment saying that the point of the code is to force a warning. Otherwise, it's pointless logic.

Reply:
It's not working, when I'm running the tests suite I can see the warning in the console (before the tests suite starts):

```
/home/guillaume/workspace/opensource/asciidoctor/asciidoctor/test/invoker_test.rb:172: warning: Ambiguous first argument; make sure.
```

And the test fails:

```
Minitest::Test::TestInvoker#test_should_enable_script_warnings_if_w_flag_is_specified [/home/guillaume/workspace/opensource/asciidoctor/asciidoctor/test/invoker_test.rb:181]:
Expected "" to not be empty.
```


```rb
  test 'should enable script warnings if -w flag is specified' do
    old_verbose, $VERBOSE = $VERBOSE, false
    begin
      warnings = nil
      redirect_streams do |out, err|
        invoke_cli_to_buffer(%w(-w -o /dev/null), '-') { String /foo/ }
        warnings = err.string
      end
      assert_equal false, $VERBOSE
      refute_empty warnings
    rescue
      $VERBOSE = old_verbose
    end
  end
```

Reply:
Never mind I get the same error with MRI:

```
/home/travis/build/asciidoctor/asciidoctor/test/invoker_test.rb:170: warning: ambiguous first argument; put parentheses or a space even after `/' operator
 ```

Not sure why but `String /foo/` is not working.

Reply:
I think it's because Ruby warns when the code is interpreted not executed.
Ruby knows that `String /foo/` is ambiguous right away but it cannot determine that `$NO_SUCH_VARIABLE` is not initialized until the code is executed.

Reply:
I'm now using:

```rb
A_CONST = 10  
A_CONST = 20
```

The above code throws a "runtime" warning on MRI, JRuby and TruffleRuby.